src/broadcast/macos/macosmediaplayerservice.mm

```diff
@@ -0,0 +1,128 @@
+        if (@available(macOS 10.13.2, *)) {
+            CoverArtCache* coverArtCache = CoverArtCache::instance();
+            CoverInfo coverInfo = pTrack->getCoverInfoWithLocation();
+            QPixmap pixmap = coverArtCache->tryLoadCover(
+                this,
+                coverInfo,
+                coverArtSize,
+                CoverArtCache::Loading::Default);
+            if (!pixmap.isNull()) {
+                setCoverArt(nowPlayingInfo, pixmap);
+            }
+        }
+
+        // Retrieve general track metadata
+        NSURL* url = [NSURL fileURLWithPath:pTrack->getLocation().toNSString()];
+        NSString* title = pTrack->getTitle().toNSString();
+        NSString* artist = pTrack->getArtist().toNSString();
+        NSString* album = pTrack->getAlbum().toNSString();
+        NSString* albumArtist = pTrack->getAlbumArtist().toNSString();
+        NSNumber* duration = @(pTrack->getDuration());
+
+        [nowPlayingInfo setObject:@(MPNowPlayingInfoMediaTypeAudio)
+                           forKey:MPNowPlayingInfoPropertyMediaType];
+        [nowPlayingInfo setObject:url forKey:MPNowPlayingInfoPropertyAssetURL];
+        [nowPlayingInfo setObject:@(false) forKey:MPNowPlayingInfoPropertyIsLiveStream];
+        [nowPlayingInfo setObject:title forKey:MPMediaItemPropertyTitle];
+        [nowPlayingInfo setObject:artist forKey:MPMediaItemPropertyArtist];
+        [nowPlayingInfo setObject:album forKey:MPMediaItemPropertyAlbumTitle];
+        [nowPlayingInfo setObject:albumArtist forKey:MPMediaItemPropertyAlbumArtist];
+        [nowPlayingInfo setObject:duration forKey:MPMediaItemPropertyPlaybackDuration];
+
+        // TODO: Use actual play time and (relative) tempo for rate
+        [nowPlayingInfo setObject:@(0.0) forKey:MPNowPlayingInfoPropertyElapsedPlaybackTime];
+        [nowPlayingInfo setObject:@(1.0) forKey:MPNowPlayingInfoPropertyDefaultPlaybackRate];
+        [nowPlayingInfo setObject:@(1.0) forKey:MPNowPlayingInfoPropertyPlaybackRate];
+
+        [center setPlaybackState:MPNowPlayingPlaybackStatePlaying];
+    }
+
+    [center setNowPlayingInfo:nowPlayingInfo];
+}
+
+void MacOSMediaPlayerService::slotCoverFound(const QObject* pRequestor,
+                                             const CoverInfo& coverInfo,
+                                             const QPixmap& pixmap,
+                                             mixxx::cache_key_t requestedImageHash,
+                                             bool coverInfoUpdated) {
+    if (pRequestor != this) {
+        return;
+    }
+
+    MPNowPlayingInfoCenter* center = [MPNowPlayingInfoCenter defaultCenter];
+    NSDictionary* existingInfo = [center nowPlayingInfo];
+    NSMutableDictionary* nowPlayingInfo =
+        existingInfo == nil
+            ? [[NSMutableDictionary alloc] init]
+            : [[NSMutableDictionary alloc] initWithDictionary:existingInfo];
+
+    // Update cover art in now playing with our found image
+    if (!pixmap.isNull()) {
```

Comment:
Since we create a new dictionary for each track, this shouldn't be an issue I think (except if the user removes his cover art from the track while playing, even then I am not sure if this function would even be called)